### PR TITLE
commit feature delete and insert user

### PR DESCRIPTION
**Pull request by namdhgc**
What i did:
 + Create function delete and insert data 
 + Create view for action delete and insert data
 + Create jQuery code for delete and insert event such as: jQuery validate, submit form using Ajax.

### CODE REVIEW

**Review by jackdiff — CHANGES_REQUESTED** [review]
Please remove unused jquery localization files.

**Comment by namdhgc** [comment]
@jackdiff i deleted unused file, please recheck.

File: app/Http/Controllers/UserController.php [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+* @since 26/12/2017 11:51:00
+*
+* @author NamDH
+*
+*/
+
+class UserController extends Controller
+{
+
+    function __construct()
```
Comment (jackdiff) [inline app/Http/Controllers/UserController.php]: please remove empty function

File: app/Http/Controllers/UserController.php [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+            if ( !empty( $avatar ) ) {
+                
+                $file_name      = $avatar->getClientOriginalName();
+                $file_extension = $avatar->getClientOriginalExtension();
+                $file_real_path = $avatar->getRealPath();
+                $file_mime      = $avatar->getMimeType();
+
+                // image mime types start with "image/"
+                if( substr( $file_mime, 0, 5 ) == 'image' ) {
+
```
Comment (jackdiff) [inline app/Http/Controllers/UserController.php:121]: Remove empty line

File: app/Http/Controllers/UserController.php [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+                $results['meta']['code']    = '0003';
+                $results['meta']['success'] = false;
+                $results['meta']['msg']     = Lang::get('message.web.error.0003');
+            }
+        }
+
+        return $results;
+    }
+
+
```
Comment (jackdiff) [inline app/Http/Controllers/UserController.php]: Should use 1 empty line

File: app/Http/Controllers/UserController.php [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+                    'value'     => $id,
+                ]
+            ];
+
+            $data = [
+                'deleted_time' => $deleted_time,
+            ];
+
+            $deleted_record = $ModelUser->update_db( $data, $where );
+            // $deleted_record['meta']['success'] = true;
```
Comment (jackdiff) [inline app/Http/Controllers/UserController.php]: remove unsed code comment out

File: routes/web.php [hunk 5]
```diff
@@ -11,6 +11,41 @@
...
+	}]);
 });
+
+Route::group(['as' => 'post-'], function () {
+
+	Route::post('delete-user', ['as' => 'delete-user', function () {
+
+		$results = App::make('App\Http\Controllers\UserController')->deleteData();
+
+		return json_encode( $results );
```
Comment (jackdiff) [inline routes/web.php]: please move return data into controller

**Review by jackdiff — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/UserController.php (hunk 1 above)
Comment (namdhgc) [reply app/Http/Controllers/UserController.php]: removed

File: app/Http/Controllers/UserController.php (hunk 3 above)
Comment (namdhgc) [reply app/Http/Controllers/UserController.php]: applied change

File: app/Http/Controllers/UserController.php (hunk 4 above)
Comment (namdhgc) [reply app/Http/Controllers/UserController.php]: removed

File: routes/web.php (hunk 5 above)
Comment (namdhgc) [reply routes/web.php]: applied change

File: app/Http/Controllers/UserController.php [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+    * Accept sort, limit, sort_type, key_search
+    *
+    * @param :
+    *
+    * @return :
+    *          + Array response
+    */
+    public function getData()
+    {
+
```
Comment (jackdiff) [inline app/Http/Controllers/UserController.php]: Remove this empty line

File: app/Http/Models/User.php [hunk 7]
```diff
@@ -0,0 +1,225 @@
...
+					break;
+				case Config::get('system.type.query.paginate'):
+
+					$query = $query->paginate($limit);
+					break;
+				default :
+					$query = $query->get();
+					break;
+			}
+			// $queries = DB::getQueryLog();
```
Comment (jackdiff) [inline app/Http/Models/User.php]: Remove these comments

File: app/Http/Controllers/UserController.php [hunk 8]
```diff
@@ -0,0 +1,215 @@
...
+            } else {
+
+                $results['meta']['code']    = '0003';
+                $results['meta']['success'] = false;
+                $results['meta']['msg']     = Lang::get('message.web.error.0003');
+            }
+        }
+
+        return json_encode( $results );
```
Comment (jackdiff) [inline app/Http/Controllers/UserController.php]: please use this convention: return response()->json($results);

File: routes/web.php [hunk 9]
```diff
@@ -11,6 +11,31 @@
...
+
+Route::group(['as' => 'get-'], function () {
+
+	Route::get('/', ['as' => 'home-page', function () {
+	    
+		return App::make('App\Http\Controllers\UserController')->getData();
+	}]);
 });
+
+Route::group(['as' => 'post-'], function () {
```
Comment (jackdiff) [inline routes/web.php]: please use this convention:
Route::group(['prefix' => 'user']), function() {
	Route::post('delete', 'UserController@deleteData');
	Route::post('add', 'UserController@insertData');
	Route::post('edit', 'UserController@updateData');
});

**Review by jackdiff — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/UserController.php (hunk 8 above)
Comment (namdhgc) [reply app/Http/Controllers/UserController.php]: applied change

File: routes/web.php (hunk 9 above)
Comment (namdhgc) [reply routes/web.php]: applied change

File: app/Http/Models/User.php (hunk 7 above)
Comment (namdhgc) [reply app/Http/Models/User.php]: removed

**Review by jackdiff — APPROVED** [review]

**Pull request merged**